src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -43,27 +43,14 @@ public class ObservableCollection<T> : Collection<T>, INotifyCollectionChanged,
         /// same order they are read by the enumerator of the collection.
         /// </remarks>
         /// <exception cref="ArgumentNullException"> collection is a null reference </exception>
-        public ObservableCollection(IEnumerable<T> collection)
+        public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection)) { }
```

Comment:
Note: I considered doing the following instead of having `CreateCopy`:

```c#
public ObservableCollection(IEnumerable<T> collection) : base(new List<T>(collection))
{
    // We let List<T>'s constructor handle throwing ArgumentNullException when collection is null
}
```

I don't have a strong preference either way.